docs/android/linter.md

```diff
@@ -0,0 +1,137 @@
+---
+title: "Android Linter"
+sidebar_label: "Linter"
+---
+
+## What is a linter?
+
+A linter is a static code analyzer that helps identify well-known issues and potential improvements in your code. It goes beyond what a compiler does by ensuring proper usage of the language and adherence to best practices. While a compiler validates code against a grammar, a linter focuses on code quality and architecture.
+
+:::note
+Having no complaints from a linter doesn't mean everything is perfect. A review from another developer is still necessary to double-check.
+:::
+
+## Why use a linter?
+
+Using a linter ensures:
+
+- **Consistency**: Enforces a standard code style, similar to our [codestyle](codestyle).
+- **Focus**: Allows reviewers to focus on logic rather than formatting or trivial issues.
+- **Prevention**: Helps avoid crashes and bugs by catching common mistakes, such as using APIs not supported by the target Android API level.
+
+For example, failing to check the Android API version before using an unsupported API can lead to crashes.
+
+## Linters used in the project
+
+### KTLint
+
+We use [KTLint](https://pinterest.github.io/ktlint) as our Kotlin linter, integrated via [Gradle plugin](https://github.com/JLLeitschuh/ktlint-gradle). The configuration is located in the main `build.gradle.kts` file. We mostly use the default configuration but enable [SARIF](tips/sarif_reports) reports for GitHub Actions to annotate issues in pull requests.
+
+#### Ignoring an issue
+
+Always try to fix issues rather than ignoring them. If ignoring is necessary, follow these steps:
+
+1. Use the `@Suppress` annotation for specific constructs:
+   ```kotlin
+   @Suppress("ktlint:standard:no-wildcard-imports")
```

Comment:
This is a bad example, as we would never accept this one.

Reply:
Very good point